### PR TITLE
include metadata in third party resource list serialization

### PR DESCRIPTION
**Pull request by sitepodmatt**
Third party resource listing does not include important metadata such as resourceVersion and apiVersion. This commit includes the missing metadata and also replaces the string templating with an anonymous struct.

### CODE REVIEW

**Comment by nowprovision** [comment]
@brendandburns this is somewhat in the same area of #25938, both are required to get up and running with third party resource clients.

**Comment by borismattijssen** [comment]
I tested this and it seems to work correctly. 
@lavalamp Can this be tested?

**Comment by lavalamp** [comment]
ok to test

Sorry, I am backlogged right now-- will look in more detail when I get a chance

File: pkg/registry/thirdpartyresourcedata/codec.go [hunk 1]
```diff
@@ -413,17 +409,38 @@ func (t *thirdPartyResourceDataEncoder) EncodeToStream(obj runtime.Object, strea
...
+		}{
+			Kind:       t.kind + "List",
+			Items:      listItems,
+			Metadata:   obj.ListMeta,
+			APIVersion: t.gv.String(),
 		}
-		fmt.Fprintf(stream, template, t.kind+"List", strings.Join(dataStrings, ","))
+
+		encBytes, err := json.Marshal(encMap)
+
```
Comment (lavalamp) [inline pkg/registry/thirdpartyresourcedata/codec.go]: omit blank line

File: pkg/registry/thirdpartyresourcedata/codec.go [hunk 2]
```diff
@@ -413,17 +409,38 @@ func (t *thirdPartyResourceDataEncoder) EncodeToStream(obj runtime.Object, strea
...
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprint(stream, string(encBytes))
```
Comment (lavalamp) [inline pkg/registry/thirdpartyresourcedata/codec.go]:
```
_, err := stream.Write(encBytes)
return err
```

**Comment by lavalamp** [comment]
This looks like a good fix. Can you add a test?

**Comment by nowprovision** [comment]
Thanks for the feedback (free golan lessons ftw :+1: ) I've implemented the changes and added a test.

**Comment by lavalamp** [comment]
Thanks. LGTM.

**Comment by gurvindersingh** [comment]
@nowprovision any update on this one.

**Comment by nowprovision** [comment]
Sorry accidentally closed this

**Comment by lavalamp** [comment]
ok to test

**Comment by goltermann** [comment]
@nowprovision most recent commit has lgtm, but failed build.

**Comment by goltermann** [comment]
@k8s-bot verify this

**Comment by idvoretskyi** [comment]
@lavalamp please, review this one.

**Comment by lavalamp** [comment]
ok to test

**Comment by goltermann** [comment]
@nowprovision vet errors are blocking this one:

Verifying ./hack/../hack/verify-govet.sh
pkg/registry/thirdpartyresourcedata/codec_test.go:259: no formatting directive in Errorf call
pkg/registry/thirdpartyresourcedata/codec_test.go:271: no formatting directive in Errorf call
pkg/registry/thirdpartyresourcedata/codec_test.go:279: wrong number of args for format in Errorf call: 1 needed but 2 args
pkg/registry/thirdpartyresourcedata/codec_test.go:283: wrong number of args for format in Errorf call: 1 needed but 2 args
exit status 1
!!! Error in ./hack/../hack/verify-govet.sh:49

**Comment by matchstick** [comment]
We should  resolve this in 1.3.x or 1.4. Moving to P0 and next-candidate.

**Comment by lavalamp** [comment]
@k8s-bot node e2e test this #IGNORE

**Pull request merged**